src/main/java/com/commercetools/sync/services/impl/CustomObjectServiceImpl.java

```diff
@@ -131,8 +134,46 @@ public CustomObjectServiceImpl(@Nonnull final CustomObjectSyncOptions syncOption
         if (StringUtils.isEmpty(customObjectDraft.getKey()) || StringUtils.isEmpty(customObjectDraft.getContainer())) {
             return CompletableFuture.completedFuture(Optional.empty());
         }
-        return createResource(customObjectDraft,
-            draft -> CustomObjectCompositeIdentifier.of(draft).toString(),
-            CustomObjectUpsertCommand::of);
+        CompletionStage<Optional<CustomObject<JsonNode>>> createdResource = createResource(customObjectDraft,
+            draft -> CustomObjectCompositeIdentifier.of(draft).toString(), CustomObjectUpsertCommand::of);
+        return createdResource ;
+    }
+
+    /**
+     * This method overrides {@link BaseService#executeCreateCommand}. It is because custom object has different
+     * behaviour from other resource in error handling, which requires to be handled separately.
+     *
+     * <p>For custom object, if without overriding, either update or create resource operations called
```

Comment:
Sadly I cannot understand what this paragraph is all about, but I tried my best and wrote something like this. It is the same meaning to your description?

>Purposefully ignoring any exception that can occur during the execution of @param createCommand Function. Any exception that occurs inside executeCreateCommand  method should delegate to the caller method which is necessary to trigger retry on error behaviour

Reply:
I got your meaning, and I think it is better to emphasize the different behaviour between custom objects and other resources. How about the following paragraph?

> Custom object has special behaviour that it only performs upsert operation. That means both update and create
> custom object operations in the end called {@link BaseService#createResource}, which is different from other
> resources.
>      
> Purposefully this method provides a specific handling to execute create command for custom objects.
> Any exception that occurs inside this method is thrown to the caller method in {@link CustomObjectSync}, which is necessary to trigger retry on error behaviour.

Reply:
And remove the word `Purposefully` 😄

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/573/commits/73bda63444316f879ab7c436391c31f8fc3ab069